eth/backend.go

```diff
@@ -559,6 +576,75 @@ func (self *Ethereum) syncAccounts(tx *types.Transaction) {
 	}
 }
 
+func (self *Ethereum) StartAutoDAG() {
+	go func() {
+		glog.V(logger.Info).Infof("Automatic pregeneration of ethash DAG ON (ethash dir: %s)", self.AutoDAGDir)
+		var nextEpoch uint64
+		ticker := time.NewTicker(autoDAGcheckInterval)
+		self.autodagquit = make(chan bool)
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
Might be a better idea to listen for `core.ChainHeadEvent` instead of waiting on a 10 hour timer.
10 hour timers are kind of hard to test.